advanced/Templates/gravity.db.sql

```diff
@@ -46,44 +87,52 @@ CREATE TABLE info
 
 INSERT INTO info VALUES("version","1");
 
-CREATE VIEW vw_gravity AS SELECT a.domain
-    FROM gravity a
-    WHERE a.domain NOT IN (SELECT domain from whitelist WHERE enabled == 1);
+CREATE VIEW vw_gravity AS SELECT domain
+    FROM gravity
+    WHERE domain NOT IN (SELECT domain from vw_whitelist);
 
-CREATE VIEW vw_whitelist AS SELECT a.domain
-    FROM whitelist a
-    WHERE a.enabled == 1
-    ORDER BY a.id;
+CREATE VIEW vw_whitelist AS SELECT domain
```

Comment:
I've tested the case where a domain appears in multiple groups, and with this code it will show up twice in the view. To fix that, we need to add `DISTINCT` after `SELECT` here.

Reply:
Confirmed:
```sql
insert into "group" (name) values ('Group 1');                                                                                                                                                                        
insert into "group" (name) values ('Group 2');                                                                                                                                                                        
insert into whitelist_by_group (whitelist_id,group_id) values (1,1);                                                                                                                                                  
insert into whitelist_by_group (whitelist_id,group_id) values (1,2);                                                                                                                                              

select * from vw_whitelist;
'whitelisted.de'
```